supervisor.py

```diff
@@ -0,0 +1,158 @@
+"""
+Red-DiscordBot built in Supervisor
+Coded by CCubed (AKA Rory in DAPI)
+"""
+import os
+import subprocess
+import sys
+import urllib.request
+from html.parser import HTMLParser
+from zipfile import *
+
+
+IS_X8664 = sys.maxsize > 2**32
+
+
+class LinkParser(HTMLParser):
+    def __init__(self):
+        self.static_download = None
+        super().__init__()
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'a':
+            for attr in attrs:
+                if attr[0] == 'href':
+                    if 'win32-static.zip' in attr[1]:
+                        self.static_download = attr[1]
+
+
+def check_for_git():
```

Comment:
That just checks to make sure that the thing in question has the platforms equivalent of +x. Not necessarily that running the file will complete successfully.